src/archanaNJun20/BookMeetingRoom.java

```diff
@@ -0,0 +1,105 @@
+package archanaNJun20;
+
+public class BookMeetingRoom {
+
+	String meetingRoom;
+	double hours=0.0;
+	boolean nalandaFlag=false;
+	boolean takshashilaFlag=false;
+	boolean agraFortFlag=false;
+	boolean pratapGadhFlag=false;
+
+	void bookMeetingRoom(){
+		if(nalandaFlag==false){
+			meetingRoom="Nalanda";
+			hours=1;
+			displayInfo();
+			nalandaFlag=true;
+		}
+		else{
+			System.out.println("-->The meeting room Nalanda is already booked.");
+		}
+	}
+	void bookMeetingRoom(String meetingRoom){
+		if(meetingRoom!="Nalanda"){
+			if(meetingRoom=="Takshashila"&&takshashilaFlag==false){
+				this.meetingRoom="Takshashila";
+				hours=1;
+				displayInfo();
+				takshashilaFlag=true;
+			}
+			else if(meetingRoom=="AgraFort" &&agraFortFlag==false){
+				this.meetingRoom="AgraFort";
+				hours=1;
+				displayInfo();
+				agraFortFlag=true;
+			}
+			else if(meetingRoom=="PratapGadh"&&pratapGadhFlag==false){
+				this.meetingRoom="PratapGadh";
+				hours=1;
+				displayInfo();
+				pratapGadhFlag=true;
+			}
+			else{
+				System.out.println(meetingRoom +"-->The meeting room Nalanda is already booked.");
+			}
+		}
+		else{
+			System.out.println(meetingRoom +"-->The meeting room Nalanda is already booked.");
+		}
+	}
+	void bookMeetingRoom(String meetingRoom,int hours){
+		if(meetingRoom!="Nalanda"){
+			if(meetingRoom=="Takshashila"&&takshashilaFlag==false){
+				this.meetingRoom="Takshashila";
+				this.hours=hours;
+				displayInfo();
+				takshashilaFlag=true;
+			}
+			else if(meetingRoom=="AgraFort"&&agraFortFlag==false){
+				this.meetingRoom="AgraFort";
+				this.hours=hours;
+				displayInfo();
+				agraFortFlag=true;
+			}
+			else if(meetingRoom=="PratapGadh"&&pratapGadhFlag==false){
+				this.meetingRoom="PratapGadh";
+				this.hours=hours;
+				displayInfo();
+				pratapGadhFlag=true;
+			}else{
+				System.out.println(meetingRoom+ "-->The meeting room  is already booked.");
```

Comment:
Here SOP should print "Invalid meeting room name" as we are not going to go inside if condition if Meeting Room name is "Nalanda"